launcher/Version.h

```diff
@@ -100,9 +106,17 @@ class Version
         int m_numPart = 0;
         QString m_stringPart;
         QString m_fullString;
+        bool m_isNull;
 
         inline bool operator!=(const Section &other) const
         {
+            if (m_isNull && other.numValid) {
+                return 0 != other.m_numPart;
+            } else if (numValid && other.m_isNull) {
+                return m_numPart != 0;
+            } else if (m_isNull || other.m_isNull) {
+                return false;
```

Comment:
oh shoot, this is my bad. 
This effectively declares all string version sections equal to a null pad (because this is implemented with inverse logic though `!=`). this should only be true for "." 

```suggestion
                if ((m_stringPart == ".") || (other.m_stringPart == ".") ) return false;
                return true;
```

Reply:
Done https://github.com/PrismLauncher/PrismLauncher/pull/751/commits/f49ad2ee03974c9fe94882d99d1a2bee67b87285